File: backend/app/schemas/overview.py
"""Overview-related schemas."""

from datetime import date, datetime

from pydantic import BaseModel, Field

from backend.app.schemas.articles import ArticleSummaryItem


class OverviewKpisPayload(BaseModel):
    """Top-level KPI payload."""

    articles_total: int
    passed_articles: int
    pending_keywords: int
    average_quality_score: float | None = None
    internal_links: int
    latest_article_at: str | None = None
    warning: str | None = None


class OverviewTrendPoint(BaseModel):
    """Single trend data point."""

    day: date | None = None
    count: int


class OverviewTrendPayload(BaseModel):
    """Trend payload."""

    days: int
    items: list[OverviewTrendPoint] = Field(default_factory=list)
    warning: str | None = None


class PendingKeywordItem(BaseModel):
    """Pending keyword card item."""

    id: int
    keyword: str
    search_volume: int
    difficulty: int


class OverviewBoardPayload(BaseModel):
    """Overview board columns."""

    pending_keywords: list[PendingKeywordItem] = Field(default_factory=list)
    draft_articles: list[ArticleSummaryItem] = Field(default_factory=list)
    ready_articles: list[ArticleSummaryItem] = Field(default_factory=list)
    warning: str | None = None


class LatestArticlesPayload(BaseModel):
    """Latest article cards."""

    items: list[ArticleSummaryItem] = Field(default_factory=list)
    limit: int
    warning: str | None = None

File: frontend_v2/src/types/overview.ts
import type { ArticleSummaryItem } from './articles'

export interface OverviewKpisPayload {
  articles_total: number
  passed_articles: number
  pending_keywords: number
  average_quality_score: number | null
  internal_links: number
  latest_article_at: string | null
  warning: string | null
}

export interface OverviewTrendPoint {
  day: string | null
  count: number
}

export interface OverviewTrendPayload {
  days: number
  items: OverviewTrendPoint[]
  warning: string | null
}

export interface PendingKeywordItem {
  id: number
  keyword: string
  search_volume: number
  difficulty: number
}

export interface OverviewBoardPayload {
  pending_keywords: PendingKeywordItem[]
  draft_articles: ArticleSummaryItem[]
  ready_articles: ArticleSummaryItem[]
  warning: string | null
}

export interface LatestArticlesPayload {
  items: ArticleSummaryItem[]
  limit: number
  warning: string | null
}
